wfe/web-front-end.go

```diff
@@ -998,11 +1008,19 @@ func (wfe *WebFrontEndImpl) Certificate(response http.ResponseWriter, request *h
 			if strings.HasPrefix(err.Error(), "gorp: multiple rows returned") {
 				wfe.sendError(response, "Multiple certificates with same short serial", err, http.StatusConflict)
 			} else {
-				wfe.sendError(response, "Not found", err, http.StatusNotFound)
+				response.Header().Add("Cache-Control", "public, max-age=0, no-cache")
+				wfe.sendError(response, "Certificate not found", err, http.StatusNotFound)
 			}
 			return
 		}
 
+		// Set cache-control header if certificate NotAfter is > time.Now().Add(-WFE.CertCacheExpirationWindow)
+		if time.Now().Add(-wfe.CertNoCacheExpirationWindow).After(cert.Expires) {
```

Comment:
I think the better approach here is to send max-age=(expires - now). @jcjones, you proposed a different approach in https://github.com/letsencrypt/boulder/issues/471, switching to no-cache shortly before the cert actually expires. Any reason for that?